compiler/cli/cli-common/src/org/jetbrains/kotlin/cli/common/messages/GroupingMessageCollector.kt

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2010-2015 JetBrains s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jetbrains.kotlin.cli.common.messages
+
+import com.google.common.collect.LinkedHashMultimap
+import com.google.common.collect.Multimap
+import kotlin.collections.sortedWith
+import org.jetbrains.kotlin.cli.common.messages.MessageCollector.Companion.reportExtended
+import java.util.Comparator
+
+open class GroupingMessageCollector(private val delegate: MessageCollector, private val treatWarningsAsErrors: Boolean) : MessageCollector {
+    // Note that the key in this map can be null
```

Comment:
This comment hints that you might have converted this declaration to Kotlin incorrectly